src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -174,6 +177,30 @@ export function DiscoverSidebar({
     return result;
   }, [fields]);
 
+  const multiFields = useMemo(() => {
+    if (!useNewFieldsApi || !fields) {
+      return undefined;
+    }
+    const map = new Map();
+    fields.forEach((field) => {
+      const parent = field.spec?.subType?.multi?.parent;
+      if (!parent) {
+        return;
+      }
+      if (!map.has(parent)) {
+        map.set(parent, []);
+      }
+      const multiField = {
+        field,
+        isSelected: selectedFields.includes(field),
+      };
+      const value = map.get(parent);
+      value.push(multiField);
+      map.set(parent, value);
```

Comment:
With the map typed properly and not to `<any, any>` TS does not longer detect that `value` mustbe set, because we set it above. Shortening that code to the following should help.

```suggestion
      const multiField = {
        field,
        isSelected: selectedFields.includes(field),
      };
      const value = map.get(parent) ?? [];
      value.push(multiField);
      map.set(parent, value);
```